### PR TITLE
Merchant adds an item

### LINKED ISSUE #45: User Story 45, Merchant adds an item
As a merchant employee
When I visit my items page
I see a link to add a new item
When I click on the link to add a new item
I see a form where I can add new information about an item, including:
- the name of the item, which cannot be blank
- a description for the item, which cannot be blank
- a thumbnail image URL string, which CAN be left blank
- a price which must be greater than $0.00
- my current inventory count of this item which is 0 or greater

When I submit valid information and submit the form
I am taken back to my items page
I see a flash message indicating my new item is saved
I see the new item on the page, and it is enabled and available for sale
If I left the image field blank, I see a placeholder image for the thumbnail

### PR DESCRIPTION
**Pull request by stillsheryl**
This PR will:
- Add functionality for merchant to add an item
- Add default image if left blank
- Update tests and model to not require image for item creation
- All tests passing
- closes #45

### CODE REVIEW

File: app/controllers/merchant/items_controller.rb [hunk 1]
```diff
@@ -5,6 +5,16 @@ def index
     @items = current_user.merchant.items
   end
 
+  def new
+  end
+
+  def create
+    merchant = current_user.merchant
+    merchant.items.create!(item_params)
```
Comment (KieraAllen) [inline app/controllers/merchant/items_controller.rb:13]: I need to remember to utilize the bang on the create method to make sure it's created correctly

File: app/views/merchant/items/index.html.erb [hunk 2]
```diff
@@ -1,10 +1,16 @@
...
 
+<%=link_to "Add a New Item", "/merchant/items/new" %>
+
 <section class="grid-container">
   <% @items.each do |item| %>
     <section class = "grid-item" id= 'item-<%=item.id%>'>
       <h2> <%=link_to item.name, "/items/#{item.id}" %> </h2>
-      <img src= <%= item.image %>>
+        <% if item.image == nil %>
+          <img id="image" src="https://thumbs.dreamstime.com/z/no-image-available-icon-photo-camera-flat-vector-illustration-132483097.jpg" >
```
Comment (KieraAllen) [inline app/views/merchant/items/index.html.erb:10]: (ONLY if we have time) we could format this picture to be a different size

File: app/views/merchant/items/new.html.erb [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+    <%= text_field_tag :name %>
+
+    <%= label_tag :description %>
+    <%= text_field_tag :description %>
+
+    <%= label_tag :image %>
+    <%= text_field_tag :image %>
+
+    <%= label_tag :price %>
+    <%= number_field_tag :price, min: 1 %>
```
Comment (KieraAllen) [inline app/views/merchant/items/new.html.erb:16]: I'm glad @c-bartell reminded us we can use min/max to limit field entry

File: db/seeds.rb [hunk 4]
```diff
@@ -36,7 +36,7 @@
 #users
 kiera = User.create!(name: 'Kiera Allen', address: '124 Main St.', city: 'Denver', state: 'CO', zip: 80205, email: 'bob@marley.com', password: 'password', role: 0)
 
-sally = User.create!(name: 'Sally Peach', address: '432 Grove St.', city: 'Denver', state: 'CO', zip: 80205, email: 'sally@peach.com', password: 'password', role: 1)
+sally = User.create!(name: 'Sally Peach', address: '432 Grove St.', city: 'Denver', state: 'CO', zip: 80205, email: 'sally@peach.com', password: 'password', role: 1, merchant_id: dog_shop.id)
```
Comment (KieraAllen) [inline db/seeds.rb:39]: Sally Peach works at Brian's Dog Shop. I've gotta go sometime! ;)

**Review by KieraAllen — APPROVED** [review]

File: app/views/merchant/items/index.html.erb (hunk 2 above)
Comment (c-bartell) [inline app/views/merchant/items/index.html.erb:10]: This is a really efficient way to accomplish this!

**Review by c-bartell — APPROVED** [review]

File: app/controllers/merchant/items_controller.rb (hunk 1 above)
Comment (JoseLopez235) [reply app/controllers/merchant/items_controller.rb:13]: second that!!

**Review by JoseLopez235 — APPROVED** [review]

**Pull request merged**